### DISCUSSION TITLE
How to create a new tenant without providing admin username & password

### DISCUSSION THREAD

**Discussion by hieucd04** [OrchardCMS/OrchardCore#11552] [Q&A]
I want to create a new tenant to host a static website (Html, CSS & JS files only).
I don't need admin user and database. However, when creating a new tenant (either from UI or from code) I couldn't find anyway to skip the providing database and admin username/password step.

Is there anyway to create a new tenant without having to provide admin username/password and database?

**Comment by hishamco** [comment]
There's an `AutoSetup` feature, but I don't think you can skip providing a database information, coz almost the things are saved into it, including users, settings, content and much more

You can create a file-based database (SQLite) to let OC save the required information, then you can start building your static website. In this case many of the OC feature will be useless but it should work

**Reply by hieucd04** [reply]
How about username & password?
I disabled the `/admin`. So, username & password are not used for anything. Can I skip providing them?

**Reply by hishamco** [reply]
The `admin` used by the administrator, you can change the admin URL prefix instead of disable it. Could you please let me know what's the benifit for using OC if the case is creating a static site without any sort of integration from OC?!!

**Reply by hieucd04** [reply]
The application I'm developing has many tenants in it.
Some tenants fully utilize OC. But some others don't.

In this very specific case, I'm tasked with adding [Storybook](https://storybook.js.org/) to the application. And the storybook has to be under a separate hostname (eg: storybook.mydomain.com). Since storybook is just a static website, there is no need for `admin` and `db`.

**Reply by hishamco** [reply]
In this case you may need to customize the setup screen or create a new recipe for static site. Frankly I didn't try it but let us hear the feedback from the others

**Reply by ns8482e** [reply]
IMO to serve static pages unique to tenant you still need to have admin

**Reply by hieucd04** [reply]
How come I need an admin while there are (from my point of view) nothing to administer?
Storybook website is auto-generated. So, the only work I can think of is deployment which is handled by our CI/CD pipeline.

**Comment by ns8482e** [comment]
Using orchard core cms, ootb Tenant can be initialized using setup / auto setup feature - this requires username and password as this will setup admin user that required by admin module

you can create your own setup module that would setup tenant without users/admin module

**Reply by hieucd04** [reply]
Good idea!!!
This might be a long term solution as my team is planning to have considerable amount of "admin-less" tenants (eg: Rest API tenants)
I will investigate it further when I have time. Thank you for the idea 😄
